Classes/Event/EnrichScriptTagEvent.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the "matomo_integration" extension for TYPO3 CMS.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ */
+
+namespace Brotkrueml\MatomoIntegration\Event;
+
+use Brotkrueml\MatomoIntegration\Exceptions\InvalidDataAttributeName;
+use Psr\Http\Message\ServerRequestInterface;
+
+/**
+ * This event allows the developer to enrich the build script tag with additional attributes.
+ */
+final class EnrichScriptTagEvent
+{
+    /**
+     * The server request in case the developer requires additional information.
+     */
+    private ServerRequestInterface $request;
+
+    /**
+     * The id of the element
+     */
+    private string $id = '';
+
+    /**
+     * The type that should be set for the script tag.
+     * e.g. text/plain to not execute script on page load, but a consent manager should handle this.
+     */
+    private string $type = '';
+
+    /**
+     * Additional data attributes.
+     * @var array <string, string|null>
+     */
+    private array $dataAttributes = [];
+
+    public function __construct(ServerRequestInterface $serverRequest)
+    {
+        $this->request = $serverRequest;
+    }
+
+    /**
+     * The server request for more context
+     */
+    public function getRequest(): ServerRequestInterface
+    {
+        return $this->request;
+    }
+
+    /**
+     * Set the id of script tag.
+     */
+    public function setId(string $id): void
+    {
+        $this->id = $id;
+    }
+
+    /**
+     * Returns the id of the script tag element
+     *
+     * @internal
+     */
+    public function getId(): string
+    {
+        return $this->id;
+    }
+
+    /**
+     * Returns the
+     *
+     * @internal
+     */
+    public function getType(): string
+    {
+        return $this->type;
+    }
+
+    public function setType(string $type): void
+    {
+        $this->type = $type;
+    }
+
+    /**
+     * Returns the data attribute array
+     *
+     * @return array<string, string|null>
+     * @internal
+     */
+    public function getDataAttributes(): array
+    {
+        return $this->dataAttributes;
+    }
+
+    /**
+     * Add a data attribute to the script tag.
+     * The prefix `data-` should be not used as part of the attribute name.
+     *
+     * @throws InvalidDataAttributeName
+     */
+    public function addDataAttribute(string $name, ?string $value = null): void
+    {
+        $name = trim($name);
```

Comment:
I would avoid trim, as it silently changes what the developer gave us. Better way would be to disallow spaces in name and throw exception because of invalid char in name.